opensearch.go

```diff
@@ -143,14 +148,20 @@ func NewDefaultClient() (*Client, error) {
 //
 // It will use http://localhost:9200 as the default address.
 //
-// It will use the ELASTICSEARCH_URL environment variable, if set,
+// It will use the OPENSEARCH_URL/ELASTICSEARCH_URL environment variable, if set,
 // to configure the addresses; use a comma to separate multiple URLs.
 //
+// It's an error to set both OPENSEARCH_URL and ELASTICSEARCH_URL.
+//
 func NewClient(cfg Config) (*Client, error) {
 	var addrs []string
 
 	if len(cfg.Addresses) == 0 {
-		addrs = addrsFromEnvironment()
+		envAddress, err := getAddressFromEnvironment()
```

Comment:
This may be a Go question, but why use envAddress as opposed to addrs directly?

Reply:
Thats correct :) 
":=" defines new variable on left side. Other alternate is

```
var err error
addrs, err = getAddressFromEnvironment()

```
Do you prefer this?

Reply:
Right, what I meant was that in Go, is this the preferred convention? 

I lean towards the approach that you just mentioned, but Im wondering if there is a convention for this?

Reply:
AFAIK, shorter declaration is preferred since it will infer type and default value by compiler. This is also preferred, if in future some one updates the return type, they don't have to update the caller as well.